File: testdata/crap/golang/sample.go
// Package sample is a fixture used by internal/analyzers/golang's tests —
// not part of the crap-metric build itself ("testdata" directories are
// ignored by the go tool).
package sample

func Simple() int {
	return 1
}

func Branchy(x int) int {
	if x > 0 {
		return x
	}
	return -x
}

func Loopy(items []int) int {
	total := 0
	for _, v := range items {
		if v > 0 && v < 100 {
			total += v
		}
	}
	return total
}

File: testdata/crap/typescript/sample.ts
// Fixture used by internal/analyzers/typescript's tests.

export function simple(): number {
  return 1;
}

export function branchy(x: number): number {
  if (x > 0) {
    return x;
  }
  return -x;
}

export function loopy(items: number[]): number {
  let total = 0;
  for (const v of items) {
    if (v > 0 && v < 100) {
      total += v;
    }
  }
  return total;
}
